nyx_client/nyx_client/configuration.py

```diff
@@ -18,59 +18,62 @@
 
 import enum
 import json
+import logging
 import os
 from dataclasses import dataclass
 
-from dotenv import dotenv_values
+from dotenv import dotenv_values, find_dotenv
+
+log = logging.getLogger(__name__)
 
 
 @dataclass(frozen=True)
 class BaseNyxConfig:
-    """Configuration for the Nyx client.
-
-    Attributes:
-        nyx_url: The URL of the Nyx instance.
-        nyx_email: The email of the Nyx user.
-        nyx_password: The password of the Nyx user.
-        override_token: Allows injection of JWT token
-    """
+    """Configuration for the Nyx client."""
 
     nyx_url: str
+    """The URL of the Nyx instance."""
     nyx_password: str
+    """The email of the Nyx user."""
     nyx_email: str
+    """The password of the Nyx user."""
     override_token: str | None = None
+    """Allows injection of JWT token"""
 
     @classmethod
     def from_env(
         cls,
-        env_file: str | None = None,
+        env_file: str = ".env",
         override_token: str | None = None,
     ):
         """Create a BaseNyxConfig instance from environment variables.
 
         Args:
-            env_file: Path to the environment file.
+            env_file: Relative (to the working directory) or absolute path to the environment file.
             override_token: Token to override the default authentication.
 
         Returns:
-            A new BaseNyxConfig instance.
+            A new `BaseNyxConfig` instance.
 
         Raises:
-            OSError: If required environment variables are not set.
+            ValueError: If a required variable is not set in the env file or the file does not exist.
         """
-        # Load from .env, note env vars will not be overwritten
-        # If no env file supplied dotenv will traverse up the directory tree looking for a .env file
-        vals = dotenv_values(dotenv_path=env_file if env_file else None)
-        url = vals.get("NYX_URL", "https://nyx-playground.iotics.space")
-        email = vals.get("NYX_EMAIL")
-        password = vals.get("NYX_PASSWORD")
-
-        if not url:
-            raise OSError("NYX_URL not set in env file")
-        if not email:
-            raise OSError("NYX_EMAIL not set in env file")
-        if not password:
-            raise OSError("NYX_PASSWORD not set in env file")
+        log.info("Specified env file: %s", env_file)
+        # NOTE: Calling find_dotenv directly instead of via dotenv_values so can use current working directory
+        if not (found_env_file := find_dotenv(filename=env_file, usecwd=True)):
+            raise ValueError(f"Env file not found from specified path: {env_file}")
```

Comment:
find_dotenv has an arg `raise_error_if_not_found` would return an IO error?

Reply:
Good point - ironically [it raises an `OSError`](https://github.com/theskumar/python-dotenv/blob/v1.0.1/src/dotenv/main.py#L316) instead of the more specific [`FileNotFoundError`](https://docs.python.org/3.10/library/exceptions.html#FileNotFoundError) which is not great either. The problem is we'd still have to try-catch to re-raise an exception specifying which file was not found as otherwise it'll be too generic an error. So I'm not sure it's worth it (compared to this check).

Also, I think it makes no sense to raise `OSError` for lack of specific options or if the file is empty (after all: it was read successfully). So maybe could do:

1. Raise `FileNotFoundError` if not found (+ update docstring to indicate)
2. Raise ValueError for the other cases

What's your view?

Reply:
Ah if the error isn't what we want then I think what you have makes more sense as it is.